feat: implemented pokemon details actions

zrnkrystnlee opened the pull request:
- added another StoreProvider in HomePage for Pokemon Details
- added a null value for Pokemon Details in AppState
- tested flutter analyze and dart format
- working results in before and after in pokemon actions; pokemon selected is Snorlax

before results
![image](https://user-images.githubusercontent.com/72370369/227070103-b8a23840-9992-403a-951b-8bec31a24cff.png)

after results
![image](https://user-images.githubusercontent.com/72370369/227070112-9fe35a15-7271-4a8a-a30f-d01b182a229a.png)

flutter analyze and dart format results
![image](https://user-images.githubusercontent.com/72370369/227070118-93ea348e-65a3-4b01-a7bf-a03f34352afa.png)

zrnkrystnlee committed:
feat: implemented pokemon details actions
- added another StoreProvider in HomePage for Pokemon Details
- added a null value for Pokemon Details in AppState
- tested flutter analyze and dart format
- working results in before and after in pokemon actions; pokemon selected is Snorlax

shawn-ffufm commented on pokedex_async_redux/lib/state/action/pokemon_actions.dart:
```diff
@@ -13,3 +13,14 @@ class GetPokemonsAction extends ReduxAction<AppState> {
+class GetPokemonDetailsAction extends ReduxAction<AppState> {
+  @override
+  Future<AppState> reduce() async {
+    final pokemonDetails = await ApiService().pokemonApi.getPokemonDetails(name: 'snorlax');
```
name should be dynamic instead of fix value, kindly utilize constructor param injection for `pokemonName` 🙂

shawn-ffufm replied:
put the snorlax value in a constant and have a todo comment as well if ever 🙂

shawn-ffufm reviewed (changes requested):
see minor pr feedback 🙂

zrnkrystnlee committed:
feat: implemented pokemon details actions
(pr feedback)
- pokemons name is now dynamic; used constructor param injection
- added pokemons name in the constants and added a TODO Comment
